combat-trainer.lic

```diff
@@ -1091,35 +1092,22 @@ class CombatTrainer
     }
     $debug_mode |= UserVars.combat_trainer_debug
     @combat_processes = make_processes(settings)
-    $target_increment = settings.combat_trainer_target_increment(3)
-    $action_count = settings.combat_trainer_action_count(10)
   end
 
   def make_processes(settings)
-    # TODO: pass all these through as one object
-    skin = settings.skinning['skin']
-    arrange_all = settings.skinning['arrange_all']
-    arrange_count = settings.skinning['arrange_count']
-    arrange_types = settings.skinning['arrange_types']
-    tie_bundle = settings.skinning['tie_bundle']
-    necro = settings.thanatology({})['necro']
-    ritual_type = settings.thanatology({})['ritual_type']
-    summoned_weapons = settings.summoned_weapons([])
-    summoned_weapons_element = settings.summoned_weapons_element(nil)
-
     [
-      SetupProcess.new(dance_threshold: settings.dance_threshold, dance_skill: settings.dance_skill, stances: settings.stances, weapon_training: settings.weapon_training({}), retreat_threshold: settings.retreat_threshold, summoned_weapons: summoned_weapons, summoned_weapons_element: summoned_weapons_element),
-      SpellProcess.new(buffs: settings.buff_spells({}), ospells: settings.offensive_spells([]), camb: settings.cambrinth, casts: settings.cast_messages, preps: settings.prep_messages, is_empath: settings.is_empath, empath_spells: settings.empath_healing),
+      SetupProcess.new(settings),
+      SpellProcess.new(settings),
       AbilityProcess.new(buffs: settings.buff_nonspells({}), kneel_khri: settings.kneel_khri(false), khri_preps: settings.khri_preps),
       LootProcess.new(
-        skin: skin,
-        arrange_all: arrange_all,
-        arrange_count: arrange_count,
+        skin: settings.skinning['skin'],
+        arrange_all: settings.skinning['arrange_all'],
+        arrange_count: settings.skinning['arrange_count'],
```

Comment:
I think we might have problems here for people who delete the skinning settings or something? Not sure but this seemed red flag to me.